From aa75d98d7e565b7a54aec62ccee6ca34f16e1f54 Mon Sep 17 00:00:00 2001
From: Osmond Chen <osmond.chen@mic.com.tw>
Date: Wed, 23 Mar 2022 12:39:48 +0800
Subject: [PATCH] Add fan presence sensor support

[PATCH] Add fan presence sensor support
[PATCH] Add fault LED setting feature supported for fan presence
[PATCH] Filter out fan presence sensor log SEL feature when service
(Fan presence side)[PATCH] Modify fan presence sensor to discrete sensor and add presence detect for tech sensor
---
 meson_options.txt         |  1 +
 service_files/meson.build |  1 +
 src/meson.build           | 18 ++++++++++++++++++
 3 files changed, 20 insertions(+)

diff --git a/meson_options.txt b/meson_options.txt
index e4e8f46..ce408ce 100755
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -12,6 +12,7 @@ option('psu', type: 'feature', value: 'enabled', description: 'Enable PSU sensor
 option('external', type: 'feature', value: 'enabled', description: 'Enable External sensor.',)
 option('event', type: 'feature', value: 'enabled', description: 'Enable Event-only sensor.',)
 option('vr-temp', type: 'feature', value: 'enabled', description: 'Enable VR temperature sensor.',)
+option('fan-presence', type: 'feature', value: 'enabled', description: 'Enable FAN presence sensor.',)
 option('tests', type: 'feature', value: 'enabled', description: 'Build tests.',)
 option('validate-unsecure-feature', type : 'feature', value : 'disabled', description : 'Enables unsecure features required by validation. Note: mustbe turned off for production images.',)
 option('insecure-sensor-override', type : 'feature', value : 'disabled', description : 'Enables Sensor override feature without any check.',)
diff --git a/service_files/meson.build b/service_files/meson.build
index a6a128a..36be5fd 100755
--- a/service_files/meson.build
+++ b/service_files/meson.build
@@ -12,6 +12,7 @@ unit_files = [
     ['external', 'xyz.openbmc_project.externalsensor.service'],
     ['event', 'xyz.openbmc_project.eventsensor.service'],
     ['vr-temp', 'xyz.openbmc_project.vrtempsensor.service'],
+    ['fan-presence', 'xyz.openbmc_project.fanpresencesensor.service'],
 ]
 
 foreach tuple : unit_files
diff --git a/src/meson.build b/src/meson.build
index 30c4d17..d5e7b2f 100755
--- a/src/meson.build
+++ b/src/meson.build
@@ -228,3 +228,21 @@ if get_option('vr-temp').enabled()
         ],
     )
 endif
+
+if get_option('fan-presence').enabled()
+    executable(
+        'fanpresencesensor',
+        'FanPresenceSensor.cpp',
+        dependencies: [
+            gpiodcxx,
+            sdbusplus,
+        ],
+        implicit_include_directories: false,
+        include_directories: '../include',
+        install: true,
+        link_with: [
+            thresholds_a,
+            utils_a,
+        ],
+    )
+endif
-- 
2.7.4

